spacy_streamlit/visualizer.py

```diff
@@ -198,10 +199,13 @@ def visualize_ner(
             default=list(labels),
             key=f"{key}_ner_label_select",
         )
+        options = {"ents": label_select, "colors": colors}
+        if extra_options:
+            options.update(extra_options)
```

Comment:
I'm trying to understand whether we'd want separate parameters for the new options instead of the generic `extra_options`. The latter is more easily extensible in the future, but it's kind of weird right now that some options are given as separate parameters (like `colors`) and others in the generic dict (like `kb_url_template`). We don't want to put them all into the generic dict because it'll break bwd-compat, but now it's kind of a mix :( What do you think?

Reply:
Ideally, I'd like to have changes in the `displacy.render` API flow seamlessly into this package - but not at the cost of breaking backward compatibility. I also understand that this change makes the API inconsistent. 

Couple of thins that come to mind:
1. Quick-fix: Introduce `kb_url_template` as a separate arg in the current API and move on. We tackle the API changes later (closer to a major version release maybe)

2. The  generic dict API is wrapped by one that does not break backward compatibility. Something like this:
```
def visualize_ner(
    doc: Union[spacy.tokens.Doc, List[Dict[str, str]]],
    *,
    ...
    colors: Dict[str, str] = {},
    options: Optional[Dict] = {},
):
    if colors:
        options["colors"] = colors
        wasabi.msg.warn("Warning message")
    _visualize_ner(doc, ..., options=options)


def _visualize_ner(
    doc: Union[spacy.tokens.Doc, List[Dict[str, str]]],
    *,
    ...
    options: Optional[Dict] = {},
) -> None:
    # THE ORIGINAL METHOD CALL WITHOUT THE `colors` arg.
```

Reply:
I think the second option is a good idea to work towards the variable `options` capturing all options in the future. Let's document the deprecation of older options in the docs, but I wouldn't throw a warning just yet, I fear it might flood people's consoles too much. We can still add a warning in a next big release, and remove the option entirely in the major release after that.

Reply:
Changes pushed. Please suggest any corrections needed in the method's docs. 

Also, about extending scope - I will make change for other visualize methods later by opening a separate ticket.